src/com/firebirdberlin/nightdream/ui/NightDreamUI.java

```diff
@@ -634,35 +710,37 @@ private void setupBackgroundImage() {
                     break;
                 }
                 case Settings.BACKGROUND_SLIDESHOW: {
-                    bgshape = loadBackgroundSlideshowImage();
+                    if (preloadBackgroundImage == null) {
+                        bgshape = loadBackgroundSlideshowImage();
+                    }
+                    else {
+                        bgshape = new BitmapDrawable(mContext.getResources(), preloadBackgroundImage);
+                        setDominantColorFromBitmap(preloadBackgroundImage);
```

Comment:
That's a possible and simple solution. Anyway I think it's better to do it async. Let's add a todo and resolve it later.
```suggestion
                        // TODO determine the dominant color in an AsyncTask
                        setDominantColorFromBitmap(preloadBackgroundImage);
```

Reply:
ok